launcher/src/main/java/net/flintmc/launcher/LaunchController.java

```diff
@@ -168,13 +168,14 @@ public void run() {
         plugin.preLaunch(rootLoader);
       }
 
-      if (launchArguments.getLaunchTarget().isEmpty()) {
-        while (true) {
-          Thread.sleep(Long.MAX_VALUE);
-        }
+      String launchTargetName = launchArguments.getLaunchTarget();
+
+      if (launchTargetName == null || launchTargetName.isEmpty()) {
+        throw new IllegalStateException(
+            "No launch target found (either set a --launch-target parameter or by using a package which specifies one)");
```

Comment:
This sounds a bit weird, should probably be `[...] (set the launch target using --launch-target or use a package which specifies one)`